.github/workflows/ci.yml

```diff
@@ -92,9 +92,12 @@ jobs:
       with:
         toolchain: stable
     - uses: Swatinem/rust-cache@v2
+    - uses: elves/setup-elvish@v1
+      with:
+        elvish-version: 0.21.0
     - name: Install shells
       if: runner.os == 'Linux'
-      run: sudo apt-get install -y elvish fish zsh
+      run: sudo apt-get install -y fish zsh
```

Comment:
A couple of concerns
- Dealing with the spread of versions of elvish that contributors will be using
- Updating `elvish-version`

We also have #5729 as another problem with there being many elvish versions out there to use

Reply:
So we'd better specify the minimal elvish version we support. This will decrease the maintaince burden I think.
BTW, I split this CI change to separate PR #5870 in case you want it.